ChangeLog.md

```diff
@@ -20,6 +20,8 @@ See docs/process.md for more on how version tagging works.
 
 3.0.1
 -----
+- Add `emscripten/thread_utils.h` helper header, which includes C++ utilities
```

Comment:
How about we add this as an internal detail of wasmfs for now and we can publicize it later if/when we think is something is widely useful?

Also, that will allow us to iterate on it while developing wasmfs without breaking any folks who start using it.

In general I think we should try to avoid creating more API surfaces that we need to support/

Reply:
Fair points, changed to be internal in wasmfs for now.

I do think this is general enough and hard enough to get right that users and we should be rewriting it. But we can consider that later.